Library/Formula/slimerjs.rb

```diff
@@ -1,3 +1,23 @@
+class FirefoxDependency < Requirement
+  fatal true
+  cask "Firefox"
+  download "https://www.mozilla.org/en-US/firefox/new"
+
+  satisfy { firefoxappdira? || firefoxappdirb? }
+
+  def firefoxappdira?
+    File.exist?("~/Applications/Firefox.app")
+  end
```

Comment:
` ~/ ` doesn't seem to work in Ruby. Using the full path works, but no dice on the ` ~/ `. ` File.exist? ` seems to be catching symlinks okay here.

```
require 'fileutils'

if File.exist?("/Users/Dominyk/Downloads/Firefox.app")
  puts "Symlinked Firefox found in downloads"
end

if File.exist?("/Users/`whoami`/Applications/Firefox.app")
  puts "Firefox found in users Applications"
end

if File.exist?("/Applications/Firefox.app")
  puts "Firefox found in global Applications"
end

```

```
Symlinked Firefox found in downloads
Firefox found in global Applications
```